Classes/Domain/Model/Changes/AbstractStructuralChange.php

```diff
@@ -50,6 +50,33 @@ abstract class AbstractStructuralChange extends AbstractChange
      */
     protected $cachedSiblingNode = null;
 
+    /**
+     * Used when creating nodes within non-default tree preset
+     *
+     * @var string|null
+     */
+    protected $baseNodeType = null;
+
+    /**
+     * Set the baseNodeType
+     *
+     * @param string $baseNodeType
+     */
+    public function setBaseNodeType($baseNodeType)
+    {
+        $this->baseNodeType = $baseNodeType;
+    }
+
+    /**
+     * Get the baseNodeType
+     *
+     * @return string|null
+     */
+    public function getBaseNodeType()
```

Comment:
```suggestion
    public function getBaseNodeType(): string
```

Reply:
Careful, It gotta be
```suggestion
    public function getBaseNodeType(): ?string
```